Kickstarter-iOS/Views/Cells/PledgeShippingLocationCell.swift

```diff
@@ -0,0 +1,106 @@
+import Library
+import Prelude
+import UIKit
+
+final class PledgeShippingLocationCell: UITableViewCell, ValueCell {
+  // MARK: - Properties
+
+  private lazy var adaptableStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private lazy var amountLabel: UILabel = { UILabel(frame: .zero) }()
+  private lazy var countryButton: UIButton = { UIButton(frame: .zero) }()
+  private lazy var titleLabel: UILabel = { UILabel(frame: .zero) }()
+  private lazy var rootStackView: UIStackView = { UIStackView(frame: .zero) }()
+  private lazy var spacer: UIView = {
+    UIView(frame: .zero)
+      |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  // MARK: - Lifecycle
+
+  override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+    super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+    _ = self
+      |> \.accessibilityElements .~ [self.titleLabel, self.countryButton, self.amountLabel]
+
+    _ = (self.rootStackView, self.contentView)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToEdgesInParent()
+
+    self.rootStackView.addArrangedSubview(self.titleLabel)
+    self.rootStackView.addArrangedSubview(self.adaptableStackView)
+
+    self.adaptableStackView.addArrangedSubview(self.countryButton)
+    self.adaptableStackView.addArrangedSubview(self.spacer)
+    self.adaptableStackView.addArrangedSubview(self.amountLabel)
+
+    self.spacer.widthAnchor.constraint(greaterThanOrEqualToConstant: Styles.grid(3)).isActive = true
+
+    self.amountLabel.setContentCompressionResistancePriority(.required, for: .horizontal)
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  // MARK: - Styles
+
+  override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> checkoutBackgroundStyle
+
+    _ = self.adaptableStackView
+      |> checkoutAdaptableStackViewStyle(self.traitCollection.ksr_isAccessibilityCategory())
+
+    _ = self.amountLabel
+      |> amountLabelStyle
+
+    _ = self.countryButton
+      |> countryButtonStyle
+      |> checkoutWhiteBackgroundStyle
+      |> checkoutRoundedCornersStyle
+
+    _ = self.countryButton.titleLabel
+      ?|> countryButtonTitleLabelStyle
+
+    _ = self.titleLabel
+      |> checkoutTitleLabelStyle
+      |> \.text %~ { _ in Strings.Your_shipping_location() }
+
+    _ = self.rootStackView
+      |> checkoutStackViewStyle
+  }
+
+  // MARK: - Configuration
+
+  func configureWith(value: (location: String, currency: String, rate: Double)) {
+    self.countryButton.setTitle(value.location, for: .normal)
```

Comment:
Should these values go into a view model? Not sure if a VM is out of the scope of this work.

Reply:
Do you mean into cell's VM? That might be out of scope and again, I'd prefer to use these cells as a thin UI layer. I'm not really sure which direction do we want to go :/